backend/src/main/java/org/cryptomator/hub/api/UsersResource.java

```diff
@@ -149,4 +156,46 @@ public List<UserDto> getAll() {
 		return userRepo.findAll().<User>stream().map(UserDto::justPublicInfo).toList();
 	}
 
+	@PUT
+	@Path("/trusted/{userId}")
+	@RolesAllowed("user")
+	@Transactional
+	@Consumes(MediaType.TEXT_PLAIN)
+	@Operation(summary = "adds/updates trust", description = "Stores a signature for the given user.")
+	@APIResponse(responseCode = "204", description = "signature stored")
+	public Response putSignature(@PathParam("userId") String userId, @NotNull String signature) {
+		var id = new WotEntry.Id();
+		id.setUserId(userId);
+		id.setSignerId(jwt.getSubject());
+		var entry = wotRepo.findById(id);
+		if (entry == null) {
+			entry = new WotEntry();
+			entry.setId(id);
+		}
+		entry.setSignature(signature);
```

Comment:
I don't think we should silently overwrite existing trusts but instead throw an 409 or something and then show to the user that there is already a trust entry and ask if the user wants to overwrite it. If the user passes a overwrite param to this endpoint it will be overwritten.

Reply:
Imo, signing data is idempotent. It makes no difference if you look at an old or a new signature, except when you no longer trust the old signature algorithm. Silently overwriting allows updating a signature, if the need arises. Also, it reduces complexity as we don't need to add new dialogs for a 409 case.